src/V3AstInlines.h

```diff
@@ -34,6 +34,13 @@ int AstNode::widthInstrs() const {
     return (!dtypep() ? 1 : (dtypep()->isWide() ? dtypep()->widthWords() : 1));
 }
 bool AstNode::isDouble() const VL_MT_STABLE {
+    if (dtypep() && VN_IS(dtypep(), UnpackArrayDType)) {
+        if (auto subDTypep = VN_AS(dtypep(), UnpackArrayDType)->subDTypep()) {
+            while (const auto nextp = subDTypep->subDTypep()) { subDTypep = nextp; }
```

Comment:
```suggestion
            while (const auto nextp = subDTypep->subDTypep()->skipRefp()) subDTypep = nextp;
```

Reply:
I'll add this to my next batch. But, can you explain the reason behind the suggestion?

Reply:
skipRefp is needed to jump over "typedef"s.    I'd suggest instead of using arrays directly in your test, make a typedef for each layer of array index instead.